node/src/service.rs

```diff
@@ -377,12 +392,30 @@ pub fn new_full<
                 let mut current_job_id: Option<String> = None;
 
                 loop {
+                    // Check for cancellation
+                    if mining_cancellation_token.is_cancelled() {
```

Comment:
Does this mean normal Ctrl-C won't work anymore? Does this intercept ctrl-c?

Reply:
no it means it stops the whole program immediately instead of waiting for it to find a block before quitting